frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -1391,6 +1391,236 @@ QualifiedType getInstantiationType(Context* context,
+                                             const EnumElement*>>;
+  ValueVector valuesAndAsts;
+  for (auto elem : enumNode->enumElements()) {
+    elem->traverse(res);
+    QualifiedType value = {};
+
+    // Found an initialization expression; use its type.
+    if (elem->initExpression()) {
+      auto qt = byPostorder.byAst(elem->initExpression()).type();
+      auto type = qt.type();
+
+      if (qt.isErroneousType() || qt.isGenericOrUnknown()) {
+        // Don't propagate errors if they're unrelated; leave value unknown.
+      } else if (!type->isIntType() && !type->isUintType()) {
+        value = CHPL_TYPE_ERROR(context, EnumInitializerNotInteger, elem, qt);
+      } else if (!qt.isParam()) {
+        value = CHPL_TYPE_ERROR(context, EnumInitializerNotParam, elem, qt);
+      } else {
+        value = qt;
+      }
+    } else {
+      if (valuesAndAsts.empty() || std::get<2>(valuesAndAsts.back()) == nullptr) {
+        // we're either the first value, or all the previous values have
+        // been abstract. We're abstract too -- encode this using a 'null'
+        // elem.
+        elem = nullptr;
+      } else {
+        auto& lastValueInfo = valuesAndAsts.back();
+        auto lastQt = std::get<0>(lastValueInfo);
+        if (lastQt.isParam()) {
+          // Previous value was valid, so add one to it.
+          value = Param::fold(context, elem, chpl::uast::PRIM_ADD, lastQt, one);
+        } else {
+          // Previous value was unknown, so we can't add one to it.
+        }
+      }
+    }
+
+    RequiredSignedness signedness = RS_NONE;
+    if (!value.param()) {
+      // Do nothing.
+    } else if (auto intParam = value.param()->toIntParam()) {
+      signedness = intParam->value() < 0 ? RS_SIGNED : RS_NONE;
+    } else if (auto uintParam = value.param()->toUintParam()) {
+      signedness = uintParam->value() > INT64_MAX ? RS_UNSIGNED : RS_NONE;
+    }
+
+    valuesAndAsts.emplace_back(value, signedness, elem);
+  }
+
+  const EnumElement* needsSigned = nullptr;
+  QualifiedType valueNeedsSigned;
+  const EnumElement* needsUnsigned = nullptr;
+  QualifiedType valueNeedsUnsigned;
+
+  for (auto& valueInfo : valuesAndAsts) {
+    auto& qt = std::get<0>(valueInfo);
+    auto signedness = std::get<1>(valueInfo);
+    auto elem = std::get<2>(valueInfo);
+    if (elem == nullptr) {
+      // abstract value; skip it.
+      continue;
+    }
+
+    if (signedness == RS_SIGNED) {
+      if (!needsSigned) {
+        needsSigned = elem;
+        valueNeedsSigned = qt;
+      }
+    } else if (signedness == RS_UNSIGNED) {
+      if (!needsUnsigned) {
+        needsUnsigned = elem;
+        valueNeedsUnsigned = qt;
+      }
+    }
+  }
+
+  bool useUnsigned = false;
+  if (needsSigned && needsUnsigned) {
+    CHPL_REPORT(context, NoTypeForEnumElem, enumNode,
+                needsSigned, valueNeedsSigned, needsUnsigned, valueNeedsUnsigned);
+  }
+  if (needsUnsigned) {
+    useUnsigned = true;
+  }
```

Comment:
It seems like ``useUnsigned`` is always equal to ``needsUnsigned``, so if we can't get rid of ``useUnsigned`` we could at least initialize it with ``needsUnsigned``. I think?

Reply:
I think it would also be useful to add a comment explaining why it's useful to proceed after the CHPL_REPORT, so that someone doesn't accidentally remove it later. Maybe a brief example of an enum that triggers this case, but from which we can still get some useful info?